src/fsharp/fsi/fsi.fs

```diff
@@ -1120,23 +1119,23 @@ type internal FsiDynamicCompiler
 
         let filteredByVal =
             allNames
-            |> List.choose (function
```

Comment:
Sure, the filter could be removed and to return non val items.  As far as I remember all the global top levels are here too like bigint etc